spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -2029,6 +2030,26 @@ public B routeToRecipients(Consumer<RecipientListRouterSpec> routerConfigurer) {
 	}
 
 	/**
+	 * Populate the {@link ErrorMessageExceptionTypeRouter} with options from the {@link RouterSpec}.
+	 * Typically used with a Java 8 Lambda expression:
+	 * <pre class="code">
+	 * {@code
+	 *  .routeByError(r -> r
+	 *      .channelMapping(IllegalArgumentException.class, "illegalArgumentChannel")
+	 *      .subFlowMapping(MessageHandlingException.class, sf ->
+	 *                                sf.handle(...))
+	 *    )
+	 * }
+	 * </pre>
+	 * @param routerConfigurer the {@link Consumer} to provide {@link ErrorMessageExceptionTypeRouter} options.
+	 * @return the current {@link IntegrationFlowDefinition}.
+	 */
+	public B routeByError(
```

Comment:
I agree, but specifically, here (and in the EMTR), we are routing on a `Throwable` payload, not an `ErrorMessage`.